components/support/sync-telemetry/src/main/java/mozilla/components/support/sync/telemetry/SyncTelemetry.kt

```diff
@@ -284,4 +288,45 @@ object SyncTelemetry {
         val message = reason.message ?: "Unexpected error: ${reason.code}"
         metric.set(message.take(MAX_FAILURE_REASON_LENGTH))
     }
+
+    fun processFxaTelemetry(jsonStr: String, crashReporter: CrashReporting? = null) {
+        val json = try {
+            JSONObject(jsonStr)
+        } catch (e: JSONException) {
+            crashReporter?.submitCaughtException(e)
```

Comment:
Nice! I found it helpful to create a separate exception class for "caught" exceptions, makes it much easier to actually filter for them in Sentry: https://github.com/mozilla-mobile/android-components/blob/master/components/support/migration/src/main/java/mozilla/components/support/migration/FennecMigrator.kt#L137

Reply:
OK, I did what I think you were suggesting :)